components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java

```diff
@@ -72,13 +75,18 @@ public void invoke(Request request, Response response) throws IOException, Servl
 
         //Get the rewrite contexts and check whether request URI contains any of rewrite contains.
         for (RewriteContext context : contextsToRewrite) {
-            Pattern pattern = context.getPattern();
-            if (pattern.matcher(requestURI).find() || pattern.matcher(requestURI + "/").find()) {
+            Pattern patternTenant = context.getTenantContextPattern();
+            Pattern patternSuperTenant = context.getBaseContextPattern();
+            if (patternTenant.matcher(requestURI).find() || patternTenant.matcher(requestURI + "/").find()) {
                 isContextRewrite = true;
                 isWebApp = context.isWebApp();
                 contextToForward = context.getContext();
                 break;
             }
+            else if (patternSuperTenant.matcher(requestURI).find() || patternSuperTenant.matcher(requestURI + "/").find()) {
+                String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+                IdentityUtil.threadLocalProperties.get().put(TENANT_NAME_FROM_CONTEXT, tenantDomain);
```

Comment:
Should not the loop broken here?

Reply:
Yes, it should. Thanks. The PR is now updated.